### ISSUE TITLE
Trigger pattern not correct

### ISSUE THREAD

**Issue by 153957** [HiSPARC/station-software#28]
The trigger patterns in the datastore are incomplete.
Only 4 bits with data are being stored, it looks like these contain the trigger pattern for the Master unit.

For a 4-detector station:
``` python
import tables
data = tables.open_file('2014_10_1.h5', 'r')
ams = data.root.hisparc.cluster_amsterdam
[bin(x) for x in set(ams.station_501.events.col('trigger_pattern'))]
```
```
['0b11000000000000000000',
 '0b00110000000000000000',
 '0b00100000000000000000',
 '0b01000000000000000000',
 '0b11110000000000000000',
 '0b10000000000000000000',
 '0b01010000000000000000',
 '0b10110000000000000000',
 '0b11010000000000000000',
 '0b11100000000000000000',
 '0b00000000000000000000',
 '0b00010000000000000000',
 '0b01110000000000000000']
```
For a 2-detector station:
``` python
 [bin(x) for x in set(ams.station_104.events.col('trigger_pattern'))]
```
```
['0b11010000000000000000',
 '0b01110000000000000000',
 '0b11110000000000000000',
 '0b01010000000000000000']
```
Zero padding was added to ensure the binary strings are all of equal length.

Since the initial commit of the OOTI Team's work the following [line](https://github.com/HiSPARC/station-software/blob/2cc0b6e4f243b66b8382fb5f9771a304385d12fa/user/hsmonitor/src/CIC.py#L70-71) has been in the CIC (event message) interpreter:
```python
self.trigger_pattern &= 2**20-1
```

This removes the 12 most significant bits from the 4 bytes ([L](https://docs.python.org/2/library/struct.html#format-characters)) that are [read out](https://github.com/HiSPARC/station-software/blob/2cc0b6e4f243b66b8382fb5f9771a304385d12fa/user/hsmonitor/src/CIC.py#L63-64). Leaving the 20 bits as shown above, with only 4 bits being used.

The data should maybe be cut the other way, keeping the 16 most significant bits and cut the 16 least significant bits.. Or there is a problem in the LabView code..

The [Message Structures HiSPARC](http://www.nikhef.nl/~hansvk/hisparc/Message_Structures_Hisparc_V4.pdf) document describes the trigger pattern to be 16 bits (2 bytes), but does not include the comparator data from the other unit (sent separately by Slave in its trigger pattern), combining the two would require 4 extra bits, hence 20 bits expected in the monitor?

**Comment by 153957** [comment]
Trigger patterns in a PySPARC 2-detector station (data from `2015_5_25.h5`):

``` python
[bin(x) for x in set(ams.station_202.events.col('trigger_pattern'))]
```
``` 
['0b00000000001000000111',
 '0b00000000001000000101',
 '0b00000000001000001101']
```

PySPARC does not modify the trigger_pattern, it simply reads the 2 bytes from the measured data message and sends that: [msg_format](https://github.com/HiSPARC/pysparc/blob/master/pysparc/messages.py#L211) -> [unpacking](https://github.com/HiSPARC/pysparc/blob/master/pysparc/messages.py#L222-L227) -> [event](https://github.com/HiSPARC/pysparc/blob/master/pysparc/events.py#L218) -> [storage](https://github.com/HiSPARC/pysparc/blob/master/pysparc/storage.py#L444)

**Comment by 153957** [comment]
It _seems_ the high thresholds are not set to -70 mV for PySPARC stations?;
```python
min(ams.station_202.events.read_where('trigger_pattern == 519')['pulseheights'][:,0])
# 1884
min(ams.station_202.events.read_where('trigger_pattern == 525')['pulseheights'][:,1])
# 1850
```
Note that the HiSPARC electronics trigger pattern does not _always_ indicate if the pulse went over the high threshold, when it triggered on the low threshold, even if it did pass the high threshold within the coincidence time window.

The high high thresholds could indicate why there is no '1111' trigger pattern, it is probably very rare.

**Comment by 153957** [comment]
HiSPARC DAQ sends the data as follows:
Trigger pattern is sent in 4 bytes, the first two are the trigger pattern from the Master (just like those sent by PySPARC), the third byte contains the slave comparator trigger pattern in its 4 least significant bits.

I propose to append those 4 slave bits to be the most significant bits in the trigger pattern.
i.e.

```
bit 0   Master Channel 1 lower threshold
bit 1   Master Channel 1 higher threshold
bit 2   Master Channel 2 lower threshold
bit 3   Master Channel 2 higher threshold
bit 4   Slave Channel 1 lower threshold
bit 5   Slave Channel 1 higher threshold
bit 6   Slave Channel 2 lower threshold
bit 7   Slave Channel 2 higher threshold
bit 8   External trigger
bit 9   Master (1) or slave (0)
bit 10  Slave present
bit 11  Master Channel 1 comparator level low
bit 12  Master Channel 1 comparator level high
bit 13  Master Channel 2 comparator level low
bit 14  Master Channel 2 comparator level high
bit 15  Calibration Mode
bit 16  Slave Channel 1 comparator level low
bit 17  Slave Channel 1 comparator level high
bit 18  Slave Channel 2 comparator level low
bit 19  Slave Channel 2 comparator level high
```

**Comment by 153957** [comment]
Should be fixed in 4a7c779

**Comment by 153957** [comment]
Much better now station 501 after some events with 4a7c779:

```
['0b11000001111',
 '0b11000010111',
 '0b11000011101',
 '0b11000110011',
 '0b11000110101',
 '0b11000111100',
 '0b11000111101',
 '0b11001000101',
 '0b11001001111',
 '0b11001010001',
 '0b11001011100',
 '0b11001011101',
 '0b11001110001',
 '0b11001110011',
 '0b11001110111',
 '0b11001111100',
 '0b11011000011',
 '0b11011000101',
 '0b11011001100',
 '0b11011001101',
 '0b11011010001',
 '0b11011110000',
 '0b11011110001',
 '0b11011110100',
 '0b11011111100',
 '0b11011111111']
```

The 11 at the front is to indicate the presence of Master and Slave. Some higher values should be appearing when there are some events with comparator messages (However, 501 has non-Nikhef PMTs so that is unlikely to happen)
